File: py/1464.py
import heapq
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        negatives = []
        heapq.heapify(negatives)

        for n in nums:
            heapq.heappush(negatives, n * - 1)

        largest = heapq.heappop(negatives) * -1

        secondLargest = heapq.heappop(negatives) * -1

        return (largest - 1) * (secondLargest - 1)

File: go/1464.go
func maxProduct(nums []int) int {
  first, second := -1, -1
  for _, num := range nums {
    if num >= first {
      first, second = num, first
    } else if num > second {
      second = num}
    }
  return (first - 1) * (second - 1)
}
